File: level1/문자열_내림차순으로_배치하기/solution.js
function solution(s) {
  var answer = s
    .split("")
    .sort((a, b) => {
      if (a.toUpperCase() === a && b.toUpperCase() !== b) return 1;
      else if (b.toUpperCase() === b && a.toUpperCase() !== a) return -1;
      return b.localeCompare(a);
    })
    .join("");
  return answer;
}

console.log(solution("Zbcdefg"));

File: level1/문자열_내림차순으로_배치하기/solution.rb
def solution(s)
  answer = s.chars.sort { |a, b|
    if (a.between?("a", "z") and b.between?("A", "Z"))
      -1
    elsif (a.between?("A", "Z") and b.between?("a", "z"))
      1
    else
      b <=> a
    end
  }.join
  return answer
end

p solution "Zbcdefg"
